pkg/operation/botanist/health_check.go

```diff
@@ -656,20 +578,23 @@ func (b *Botanist) checkSystemComponents(
 	ctx context.Context,
 	checker *HealthChecker,
 	condition gardencorev1beta1.Condition,
-	shootDeploymentLister kutil.DeploymentLister,
-	shootDaemonSetLister kutil.DaemonSetLister,
 	extensionConditions []ExtensionCondition,
 ) (*gardencorev1beta1.Condition, error) {
 
-	if exitCondition, err := checker.CheckSystemComponents(metav1.NamespaceSystem, condition, shootDeploymentLister, shootDaemonSetLister); err != nil || exitCondition != nil {
-		return exitCondition, err
-	}
-	if exitCondition, err := checker.CheckMonitoringSystemComponents(metav1.NamespaceSystem, b.Shoot.GetPurpose() == gardencorev1beta1.ShootPurposeTesting, condition, shootDeploymentLister, shootDaemonSetLister); err != nil || exitCondition != nil {
-		return exitCondition, err
-	}
-	if exitCondition, err := checker.CheckOptionalAddonsSystemComponents(metav1.NamespaceSystem, condition, shootDeploymentLister, shootDaemonSetLister); err != nil || exitCondition != nil {
-		return exitCondition, err
+	for name := range common.ManagedResourcesShoot {
+		mr := &resourcesv1alpha1.ManagedResource{}
+		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, name), mr); err != nil {
+			return nil, err
+		}
+
+		if mr.Generation != mr.Status.ObservedGeneration {
+			return nil, fmt.Errorf("observed generation of managed resource %s/%s outdated (%d/%d)", mr.Namespace, mr.Name, mr.Status.ObservedGeneration, mr.Generation)
+		}
```

Comment:
What's the reason, you didn't put the generation check into the `CheckManagedResource` func?

Reply:
Because the surrounding function handles the error with an unkown condition. I thought that would be more appropriate.

Reply:
Hmm, `Unknown` is usually only used we cannot determine the status of a health check, for example when we cannot talk to the Kubernetes API server. If we detect an outdated condition then we can indeed determine the status and should probably report a failed condition.